### ISSUE TITLE
why h_size is = 512?

### ISSUE THREAD

**Issue by ToutaF** [gcamfer/Anomaly-ReactionRL#21]
Hello,
I did'nt understand some of the code, can you please explain to me?
` def __init__(self,scope="estimator",h_size=512,summaries_dir=None):    
        self.scope = scope
        self.h_size = h_size
        # Writes Tensorboard summaries to disk
        self.summary_writer = False
        with tf.variable_scope(scope):
            # Build the graph
            self._build_model()
            if summaries_dir:
                summary_dir = os.path.join(summaries_dir, "summaries_{}".format(scope))
                if not os.path.exists(summary_dir):
                    os.makedirs(summary_dir)
                self.summary_writer = tf.summary.FileWriter(summary_dir)
        
    def _build_model(self):   
        #The network recieves a frame from the game, flattened into an array.
        #It then resizes it and processes it through four convolutional layers.
        self.Input =  tf.placeholder(shape=[None,env.obs_size],dtype=tf.float32)
        
        
        self.out1 = layer.fully_connected(inputs=self.Input,num_outputs=100)
        self.out2 = layer.fully_connected(inputs=self.out1,num_outputs=100)
        self.out3 = layer.fully_connected(inputs=self.out2,num_outputs=self.h_size)
        
        #We take the output from the final convolutional layer and split it into separate advantage and value streams.
        self.streamAC,self.streamVC = tf.split(self.out3,2,1)`

1- why h_size=512? isn't sould be =5 (number of classes)?
2- why you split the output of the output layer into advantage and value streams?

**Comment by gcamfer** [comment]
Hi!!

Sorry for the late.

The h_size is the hidden size of the duelling network.
The layer just before the break of the value and advantage.
This is not the final action

![image](https://user-images.githubusercontent.com/11620583/112113719-7699d080-8bb7-11eb-9a87-4eb4094c9838.png)

I hope I have explained it correctly.
Regards

**Comment by ToutaF** [comment]
thank you for your response. why  it had 512 nodes e.g why this value?.
another question please. the code gives an accurcy rate =74% how can I improve it

**Comment by gcamfer** [comment]
It have 512 as a default value, it's not necesary to be 512.
As is usual, powers of 2 are used for the dimensions of neural networks in order to make the calculations easier on CPUs.
I didn't remmember well how to obtain better acc but some of hyperparameters you can choose are size of hidden, deep (nº hidden layers), gamma, learning rates and optimicers as Adam, adagrad, etc

**Comment by ToutaF** [comment]
ok.
What is the difference between gamma (the discount factor) and decay rate please?

**Comment by gcamfer** [comment]
The decay rate involves the policy strategy in order to select best actions with a starting point of randomnes and dacaying to get only most valuable actions.

Gamma involves the marcov steps to get info from previous Qvalues so in this case gamma sould be really low because of the independence of the states.

**Comment by ToutaF** [comment]
you provided many solutions for the intrusion problem. If I have to choose between creating an agent with tf agents (gym) or creating the agent without gym, What do you recommend in the case of intrusion detection?

**Comment by gcamfer** [comment]
Hi, I am sorry to reply so late, 
I had missed the open issues in the repository as I am not actively working on it.
I would choose tf agents if it is a known one and if developing one from scratch clearly one of my own.
I have another repository where its' implemented the environment for classification with a compatible gym environment [gym-classification](https://github.com/gcamfer/gym-classification)
Close the issue as this is not the right place for questions.

Regards.
